packages/node_modules/pouchdb-find/src/adapters/local/abstract-mapper.js

```diff
@@ -12,40 +12,38 @@ import { matchesSelector, parseField } from 'pouchdb-selector-core';
 // the function, but it would also be a lot less performant.
 //
 
+function getDeepValue(value, path) {
+  for (const key of path) {
+    value = value[key];
+    if (value === undefined) {
+      throw new Error("value is undefined");
```

Comment:
This was the best idea I came up to avoid having another loop, checking for `undefined` in createDeepMultiMapper.

Can you imagine a better solution?
I would be happy to implement it for you 😉.

Reply:
Ahh I see... in createDeepMultiMapper you don't want it to emit any fields if any are undefined. And because you're using `.map` this is hard.

In that case I suggest replacing the `map` in `createDeepMultiMapper` with a for loop so you can `return` when necessary. Something like...

```js
    const toEmit[];
    for (const field of fields) {
      const value = getDeepValue(doc, parseField(field));
      if (value === undefined) {
        return;
      }
      toEmit.push(value);
    }
    emit(toEmit);
```

Reply:
OK, now it's back to loop.
Just as it was before I tried to follow this advise: https://github.com/pouchdb/pouchdb/pull/8860#discussion_r1544643471 😂